### PR TITLE
Combine slateruhf and slaterpbc into one object

### PR DESCRIPTION
**Pull request by willwheelera**
slateruhf.py and slaterpbc.py are replaced by a new slater.py. Some helper functions are added where pbc and mol implementations differ so that the Slater object can adopt the appropriate function. I used an if statement in pgradient, but can also replace that with helper functions if that's better.

I also updated the tests and __init__ files to use the new combined Slater object.

### CODE REVIEW

File: pyqmc/slater.py [hunk 1]
```diff
@@ -1,100 +1,111 @@
...
+        self._coefflookup = ("mo_coeff_alpha", "mo_coeff_beta")
+
+        if hasattr(mol, "a"):
+            self._init_pbc(mol, mf, twist)
+        else:
+            self._init_mol(mol, mf)
+        self.pbc_str = "PBC" if hasattr(mol, "a") else ""
+        self._aostack = _aostack_pbc if hasattr(mol, "a") else _aostack_mol
+
+        print("scf object is type", type(mf))
```
Comment (lkwagner) [inline pyqmc/slater.py:52]: Please remove stray print statements in __init__()

File: pyqmc/slater.py [hunk 2]
```diff
@@ -1,100 +1,111 @@
...
             twist = np.zeros(3)
         else:
-            twist = np.dot(np.linalg.inv(supercell.a), np.mod(twist, 1.0)) * 2 * np.pi
-        self._kpts = get_supercell_kpts(supercell) + twist
+            twist = np.dot(np.linalg.inv(cell.a), np.mod(twist, 1.0)) * 2 * np.pi
+        self._kpts = get_supercell_kpts(cell) + twist
         kdiffs = mf.kpts[np.newaxis] - self._kpts[:, np.newaxis]
         self.kinds = np.nonzero(np.linalg.norm(kdiffs, axis=-1) < 1e-12)[1]
         self.nk = len(self._kpts)
         print("nk", self.nk, self.kinds)
```
Comment (lkwagner) [inline pyqmc/slater.py:103]: Another print statement.

File: pyqmc/slater.py (hunk 1 above)
Comment (willwheelera) [reply pyqmc/slater.py]: done

File: pyqmc/slater.py (hunk 2 above)
Comment (willwheelera) [reply pyqmc/slater.py]: done

**Pull request merged**